### ISSUE TITLE
error[unresolved-reference] nested functions false positive

### ISSUE THREAD

**Issue by jc-louis** [astral-sh/ty#389]
### Summary

Hello,
I'm having false positive with `error[unresolved-reference]` in a very specific case of nested functions.

Minimal example:
```py
def bug_inner() -> str:
    if True: # no error without this if
        inner = "x"

        def _inner_fct() -> str:
            if "x" == inner:
                return "x"

        return _inner_fct()
```

The error:
```
error[unresolved-reference]: Name `inner` used when not defined
 --> scratch_1425.py:6:23
  |
5 |         def _inner_fct() -> str:
6 |             if "x" == inner:
  |                       ^^^^^
7 |                 return "x"
  |
```

### Version

ty 0.0.1-alpha.1 (12f466e46 2025-05-13)

**Comment by carljm** [comment]
Thanks for the report! This is a useful concise example case for https://github.com/astral-sh/ty/issues/210 -- I think I will close it as a duplicate, but it's very useful, thank you.
